src/Model/ReferencesTrait.php

```diff
@@ -80,7 +80,7 @@ public function addRef(string $link, array $defaults): Reference
      *
      * @return Reference\HasOne
      */
-    public function hasOne(string $link, array $defaults = []): Reference
+    public function hasOne(string $link, array $defaults = []) //: Reference
```

Comment:
otherwise autohinting (from phpdoc) does not work

Reply:
Why not add HasOne to Method Signature? 
```
public function hasOne(string $link, array $defaults = []): Reference\HasOne
```

Reply:
because it can be implemented by another class not based on this one

Reply:
so you mean a child class could e.g. implement 
`public function hasOne(string $link, array $defaults = []): Some\Other\Reference`
and that would fail if we set the parent signature to Reference\HasOne?

Reply:
yes

Reply:
grml this is ugly. Do you really think adding `Reference\HasOne` would be an issue? I see return types in method signatures as an important part of good code. While this PR helps IDEs and phpstan and so on using comments, it weakens the method signatures which was something I liked about almost any of your PR: You created stricter, clearer method signatures.